### DISCUSSION TITLE
How does one use the vim mode in marimo?

### DISCUSSION THREAD

**Discussion by phubaba** [marimo-team/marimo#2224] [General]
I have enabled vim keybindings. 

When i press escape for example to get to normal mode, i get pulled out of the marimo cells entirely and none of the keyboard short cuts work at all. (like the ctrl+shift+j/k start interacting with the browser). there doesnt seem to be a way to get back to the cells without clicking.  When Im in this mode j/k does move the notebook up and down but without a way to select a cell im at a loss on how to use the vim mode.

**Comment by akshayka** [comment]
@phubaba 

Sorry it's not quite working for you!

I am unable to reproduce your issue. Here is a video of escape not pulling me out of the cells, and me being able to focus back into the cell without using the mouse. 

[vim-mode.webm](https://github.com/user-attachments/assets/2340cbd3-12ff-4b20-819b-d18d86e0bf69)

If you watch the video to the end, the last comment is supposed to say "without using the mouse" 😅

Can you share more information to help us debug, such as a video or reproduction, information about what browser you are using, what marimo version, etc.?

**Reply by phubaba** [reply]
thanks for verifying that it works on your end. that motivated me to try other browsers. edge does actually work properly! cheers

Im on marimo 0.8.10, chrome version 128.0.6613.120 (Official Build) (64-bit). 

I dont have a way to share a video (this is happening on a computer i cant install any windows apps onto).

**Reply by phubaba** [reply]
The best way to describe it in chrome is

1. I click on the cell. It enters as normal mode. 
2. I then press i and start typing. 
3. I press escape to enter normal mode, but instead it sends me out to a mode where no cell is selected and j/k just scroll the notebook, I'll note this mode is very similar to what happens (even in edge) when you say click on the output of an accordion cell. No keyboard commands work in that case and you have to click back onto a cell. 
4. If I then click on the same cell i am still in insert mode. 
5. I can click another cell and i will enter in normal mode, but then if i press i and then escape that cell will act like above
